File: js/palindrome.js
exports.palindrome = function(word) {
    // ensure word is a string
    word = word.toString();
    // globally (/g) replace/remove chars not a-z, A-Z or 0-9	
    let tmpWord = word.replace(/[^a-zA-Z0-9]/g, '');
    // convert to lower case
    string = tmpWord.toLowerCase();
    // a single letter is technically a palindrome
    if (string.length === 1){	
      return true;	
    }
    // call the helper function to check if string is equal to its reverse	
    return (string === revString(string));	
};    
// helper function     
function revString(str){	
    strArr = str.split('');	
    arrRev = strArr.reverse();	
    return arrRev.join('');	
}

// console.log(exports.palindrome('nice'))

File: python/palindrome.py
def palindrome(word):
    # needed for re.sub
    import re
    # ensure that word is string type
    word = str(word)
    # remove non-alphannumeric characters
    string = re.sub(r"[^a-zA-Z0-9]", "", word)
    # convert to lower case
    string = string.lower()
    # a single letter is technically a palindrome 
    if len(string) == 1:
        return true
    # call the helper function to check if string is equal to its reverse
    #print(rev_string(string))
    return (string == rev_string(string)) 

def rev_string(str):
    str_arr = list(str)
    str_arr.reverse()
    return ''.join(str_arr)

print(palindrome('434'))